src/renderers/native/ReactNativeAttributePayload.js

```diff
@@ -314,6 +318,9 @@ function diffProperties(
   for (var propKey in nextProps) {
     attributeConfig = validAttributes[propKey];
     if (!attributeConfig) {
+      if (__DEV__ && enableStrictAttributeValidation === true && propKey !== 'children') {
+        throw new Error('unsupported attribute: ' + propKey);
```

Comment:
Using a warning makes sense to me, fixed.

I also expect that this will expose warnings on existing native apps where some invalid attributes currently no-op. There may even be some cases where someone wants to use invalid attributes and no-op (e.g. if a component's JSX refers to a subcomponent that gets swapped out for another subcomponent that only supports a subset of the original's attributes). However, I think in most cases it's nice to get a warning that the attribute you're trying to set is invalid (adding this warning has already helped me catch one such case). I originally suggested making the enableStrictAttributeValidation property configurable so someone can toggle it without having to rebuild React, but I didn't see an existing way to support configuration without adding a new top-level property to React, which is overkill just for this single property. It would be nice to have a React.configure() mechanism. Thoughts?